### PR TITLE
Add Traditional Chinese translation for spec 2.0.0

### PR DESCRIPTION
**Pull request by wmh**

### CODE REVIEW

**Comment by haacked** [comment]
Thanks! I don't know Chinese so I can't really review this. Do you know anyone else willing to review it?

**Comment by muan** [comment]
:exclamation: I will totally review this. :blush:

**Comment by haacked** [comment]
@muan :thumbsup: Thanks!

**Comment by muan** [comment]
Hey also @dlackty, you'd probably do a better job than me with this.

File: lang/zh-TW/index.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+1. 修訂號 Z（x.y.Z | x > 0）「必須 `MUST`」在只做了向下相容的修正時才遞增。這裡
+的修正指的是針對不正確結果而進行的內部修改。
+
+1. 次版號 Y（x.Y.z | x > 0）「必須 `MUST`」在有向下相容的新功能出現時遞增。在任
+何公共 API 的功能被標記為棄用時也「必須 `MUST`」遞增。也「可以 `MAY`」在私有程式
+內有大量新功能或改進被引入時遞增，其中「可以 `MAY`」包括修訂級別的改變。每當次版
+號遞增時，修訂號「必須 `MUST`」歸零。
+
+1. 主版本号 X（X.y.z | X > 0）「必須 `MUST`」在有任何不相容的修改被引入到公共
```
Comment (dlackty) [inline lang/zh-TW/index.md]: `号` could be replaced with `號` to be consistent with other paragraphs.

File: lang/zh-TW/index.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+當你的軟體被用於正式環境，就可以算是 1.0.0 版。如果你已經有個穩定的 API 被使用者
+依賴，也會是 1.0.0 版。如果你很擔心向下相容的問題，也應該算是 1.0.0 版了。
+
+### 這不會阻礙快速開發和迭代嗎？
+
+主版號為零是完全為了快速開發。如果你每天都在改變 API，那麼你應該仍在 0.y.z 版或
+是正在下個主版本的獨立開發分支中。
+
+### 對於公共 API，若即使是最小但不向下相容的改變都需要產生新的主版號，豈不是很快
+就達到 42.0.0 版？
```
Comment (dlackty) [inline lang/zh-TW/index.md]: I guessed this should be the same line of the heading line above?

File: lang/zh-TW/index.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+### 語意化版本對於版本的字串長度是否有限制呢？
+
+沒有，請自行做適當的判斷。舉例來說，長到 255 個字元的版本已過度誇張。再者，特定
+的系統對於字串長度可能會有他們自己的限制。
+
+關於
+----
+
+語意化版本控制的規範是由 Gravatars 創辦者兼 Github 共同創辦者 [Tom
```
Comment (dlackty) [inline lang/zh-TW/index.md]: It is `GitHub` right? @muan ?

**Comment by dlackty** [comment]
@wmh Really nice work! 
Actually I used to try to translate it but didn't finish, so it's exciting to see this happen.
Can't wait to share this with other Chinese developers. :+1:

**Comment by wmh** [comment]
@dlackty Thanks for your kindly help. I really appreciate it. :bow:

File: lang/zh-TW/index.md (hunk 3 above)
Comment (muan) [inline lang/zh-TW/index.md]: :+1::+1::+1::sparkles: haha

Comment (wmh) [inline lang/zh-TW/index.md]: @dlackty :+1: :+1: :+1: Good eyesight!

File: lang/zh-TW/index.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+================
+
+摘要
+----
+
+版本格式：主版號.次版號.修訂號，版號遞增規則如下：
+
+1. 主版號：當你做了不相容的 API 修改，
+1. 次版號：當你做了向下相容的功能性新增，
+1. 修訂號：當你做了向下相容的問題修正。
```
Comment (muan) [inline lang/zh-TW/index.md]: Although it'd be the same after rendering, I think it'd make more sense to make it 1. 2. 3. ..

File: lang/zh-TW/index.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+先行版號及編譯版號可透過額外的符號被附加到「主版號.次版號.修訂號」的格式後面。
```
Comment (muan) [inline lang/zh-TW/index.md]: 「透過額外的符號」感覺有點模糊、不明意義，是不是可以省略掉？
「先行版號及編譯版號可以加到「主版號.次版號.修訂號」的後面，作為延伸。」？

File: lang/zh-TW/index.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+簡介
+----
+
+在軟體管理的領域裡存在著被稱作「相依性地獄」的死亡之谷，系統規模越大，引入的套件
+越多，你就越有可能在未來的某一天發現自己已深陷絕望之中。
+
+在相依性高的系統中發佈新版本套件可能很快會成為惡夢。如果相依性關係過高，可能面臨
+版本控制被鎖死的風險（必須對每一個相依套件改版才能完成某次升級）。而如果相依性關
+係過於鬆散，又將無法避免版本的混亂（假設相容於未來的多個版本已超出了合理數量）。
+當你專案的進展因為版本相依被鎖死或版本混亂變得不夠簡便和可靠，也就意味著你正處於
```
Comment (muan) [inline lang/zh-TW/index.md]: 刪除「也」？

File: lang/zh-TW/index.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+相依性地獄之中。
+
+作為這個問題的解決方案之一，我提議用一組簡單的規則及需求來約束版號的分配和增長。
```
Comment (muan) [inline lang/zh-TW/index.md]: 「需求」翻成「條件」應該比較順，另外「配置(n)」比「分配(v)」適合。

File: lang/zh-TW/index.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+這些規則是基於（但不局限於）已廣泛被實做的封閉及開放源碼軟體。為了讓這套理論運
```
Comment (muan) [inline lang/zh-TW/index.md]: 「這些規則是根據（但不局限於）已經被各種封閉、開放源碼軟體所廣泛使用的慣例所設計。」
不確定這樣有沒有比較好，但是「基於」和「實作」在這邊用感覺怪怪的。

File: lang/zh-TW/index.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+作，你必須先有定義好的公共 API。這可以透過文件定義或程式碼強制要求來實現。無論如
+何，這套 API 的清楚明瞭是十分重要的。一旦你定義了公共 API，你就可以透過修改相應
+的版號來告知大家你的修改。考慮使用這樣的版號格式：X.Y.Z（主版號.次版號.修訂號）
```
Comment (muan) [inline lang/zh-TW/index.md]: 「告知」在這裡不太precise，因為不只是讓大家知道有修改，還有版本號所附加的意義在。我覺得「說明」會比較好。

File: lang/zh-TW/index.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+以下關鍵詞 MUST、MUST NOT、REQUIRED、SHALL、SHALL NOT、SHOULD、SHOULD NOT、
+RECOMMENDED、MAY、OPTIONAL 依照 RFC 2119 的敘述解讀。（譯著：為了保持語句順暢，
+以下文件遇到的關鍵詞將依照整句語意進行翻譯，在此先不進行個別翻譯。）
+
+1. 使用語意化版本控制的軟體「必須 `MUST`」定義公共 API。該 API 可以在程式碼中被
+定義或出現於嚴謹的文件內。無論何種形式都應該力求精確且完整。
+
+1. 標準的版號「必須 `MUST`」採用 X.Y.Z 的格式，​​其中 X、Y 和 Z 為非負的整數，且
+「禁止 `MUST NOT`」以 0 為開頭。X 是主版號、Y 是次版號、而 Z 為修訂號。每個元素
```
Comment (muan) [inline lang/zh-TW/index.md]: 「以 0 為開頭」有點怪，我會覺得是不是主版號不能為 0，說「禁止在數字前方補零」會不會比較好？

File: lang/zh-TW/index.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+API 不應該被視為穩定版。
+
+1. 1.0.0 的版號用於界定公共 API 的形成。這一版本之後所有的版號更新都基於公共 API
+及其修改內容。
+
+1. 修訂號 Z（x.y.Z | x > 0）「必須 `MUST`」在只做了向下相容的修正時才遞增。這裡
+的修正指的是針對不正確結果而進行的內部修改。
+
+1. 次版號 Y（x.Y.z | x > 0）「必須 `MUST`」在有向下相容的新功能出現時遞增。在任
+何公共 API 的功能被標記為棄用時也「必須 `MUST`」遞增。也「可以 `MAY`」在私有程式
```
Comment (muan) [inline lang/zh-TW/index.md]: 我覺得「私有程式」怪怪的，但是說「內部程式」又怕被當做 API 的程式碼...有沒有什麼其他的選項呀？

File: lang/zh-TW/index.md [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+內有大量新功能或改進被引入時遞增，其中「可以 `MAY`」包括修訂級別的改變。每當次版
+號遞增時，修訂號「必須 `MUST`」歸零。
+
+1. 主版本號 X（X.y.z | X > 0）「必須 `MUST`」在有任何不相容的修改被引入到公共
```
Comment (muan) [inline lang/zh-TW/index.md]: 「引入到公共API」太不口語了..「加入公共API」怎麼樣？

File: lang/zh-TW/index.md [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+API 時遞增。其中「可以 `MAY`」包括次版號及修訂級別的改變。每當主版號遞增時，次版
+號和修訂號「必須 `MUST`」歸零。
+
+1. 先行版號「可以 `MAY`」被標注在修訂版之後，先加上一個破折號再加上一連串以句點
```
Comment (muan) [inline lang/zh-TW/index.md]: 應該是「連接號」不是「破折號(em dash)」

File: lang/zh-TW/index.md (hunk 5 above)
Comment (muan) [inline lang/zh-TW/index.md]: 另外 build metadata 翻作編譯版號感覺不太恰當，畢竟他不是「版號」而是...metadata..如果可能是SHA，那就更不是版號了。「版本資訊」如何？

File: lang/zh-TW/index.md [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+
+為什麼要使用語意化的版本控制？
+------------------------------
+
+這並不是一個新的或者革命性的想法。實際上，你可能已經在做一些近似的事情了。問題在
+於只是「近似」還不夠。如果沒有某個正式的規範可循，版號對於相依性的管理並無實質意
+義。將上述的想法命名並給予清楚的定義，讓你對軟體使用者傳達意向變得容易。一旦這些
+意向變得清楚，彈性（但又不會太彈性）的相依性規範就能達成。
+
+舉個例子來展示語意化的版本控制如何讓相依性地獄成為過去。假設有個名為「救火車」的
```
Comment (muan) [inline lang/zh-TW/index.md]: 「舉個簡單的例子就可以展示...」

File: lang/zh-TW/index.md [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+函式庫依賴另一個已使用語意化版本控制，名為「梯子」的套件。當救火車創建時，梯子的
```
Comment (muan) [inline lang/zh-TW/index.md]: 「...名為「救火車」的函式庫，它需要另一個名為「梯子」並已經有使用語意化版本控制的套件。」

File: lang/zh-TW/index.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+版號為 3.1.0。因為救火車使用了一些版本 3.1.0 所新增的功能，你可以放心地指定相依
+於梯子的版號大等於 3.1.0 但小於4.0.0。這樣，當梯子版本 3.1.1 和 3.2.0 發佈時，你
+可以將直接它們納入你的套件管理系統，因為它們能與原有相依的軟體相容。
+
+作為一位負責任的開發者，你理當確保每次套件升級的運作與表述一致。現實世界是複雜
```
Comment (muan) [inline lang/zh-TW/index.md]: 「的運作與 **版本號的** 表述一致」

File: lang/zh-TW/index.md [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+的，我們除了提高警覺外能做的不多。你所能做的就是讓語意化的版本控制為你提供一個健
+全的方式來發行以及升級套件，而無需推出新的相依套件，節省你的時間及煩惱。
+
+如果你對此認同，希望立即開始使用語意化版本控制，你只需聲明正在使用它並遵循這些規
```
Comment (muan) [inline lang/zh-TW/index.md]: 「你只需聲明 **你的函式庫** 正在使用它 **、** 並遵守..」

File: lang/zh-TW/index.md [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+FAQ
+---
+
+### 在 0.y.z 初始開發階段，我該如何進行版本控制？
+
+最簡單的做法是以 0.1.0 作為你的初始化開發版本，並在後續的每次發行時遞增次版號。
+
+### 如何判斷發佈 1.0.0 版本的時機？
+
+當你的軟體被用於正式環境，就可以算是 1.0.0 版。如果你已經有個穩定的 API 被使用者
```
Comment (muan) [inline lang/zh-TW/index.md]: 「當你的軟體被用於正式環境，它應該已經達到了 1.0.0 版。」
畢竟被別人使用是自己無法控制的，我剛開發的第一天就可能有人拿去 prod 用，但這不代表我已經到了 1.0.0。

File: lang/zh-TW/index.md [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+依賴，也會是 1.0.0 版。如果你很擔心向下相容的問題，也應該算是 1.0.0 版了。
+
+### 這不會阻礙快速開發和迭代嗎？
+
+主版號為零是完全為了快速開發。如果你每天都在改變 API，那麼你應該仍在 0.y.z 版或
```
Comment (muan) [inline lang/zh-TW/index.md]: 「主版號為零 **的時候** 就完全是為了做快速開發。... 那麼你應該仍在 **主版號為零的階段（0.y.z），** 或是...」

File: lang/zh-TW/index.md [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+
+為供他人使用的軟體編寫適當的文件，是你作為一名專業開發者應盡的職責。保持專案高效
+一個非常重要的部份是掌控軟體的複雜度，如果沒有人知道如何使用你的軟體或不知道哪些
+函數的呼叫是可靠的，要掌控複雜度會是困難的。長遠來看，使用語意化版本控制以及對於
+公共 API 有良好規範的堅持，可以讓每個人及每件事都運行順暢。
+
+### 萬一不小心把一個不相容的改版當成了次版號發行了該怎麼辦？
+
+一旦發現自己破壞了語意化版本控制的規範，就要修正這個問題，並發行一個新的次版號來
+更正這個問題並且回復向下相容。即使是這種情況，也不能去修改已發行的版本。可以的
```
Comment (muan) [inline lang/zh-TW/index.md]: 雖然「回復」應該同意，但我覺得使用「恢復」會比較好理解。

**Comment by muan** [comment]
Overall :+1:!! 

There are a few places that I feel jargons are used unnecessarily, which makes them not that easy to read. I am nitpicking a bit, but just to be honest the English version Tom wrote was a lot easier to understand. There are a couple passive/active sentences that would be a lot more understandable if changed into active/passive sentences in Chinese.

Naming wise, I asked @dlackty cause he's probably read a lot more programming books in Chinese than me, there are names I felt difficult to understand, but if that's the normal Chinese programming term, ignore those corrections. :wink: 

Again, really good job. I can never translate it all by myself. :)

File: lang/zh-TW/index.md (hunk 5 above)
Comment (wmh) [inline lang/zh-TW/index.md]: build metadata 這一詞似乎不常見，我用「編譯版號」會和其他幾個版號的詞有比較一致的感覺。

我看了一下目前全文有五處用到這個詞，都改用「編譯版本資訊」是可以通順的。

@dlackty 有什麼建議嗎？

File: lang/zh-TW/index.md (hunk 10 above)
Comment (wmh) [inline lang/zh-TW/index.md]: :+1: 後面有一段提到先行版號的部份也有相同問題，一併修改了。

File: lang/zh-TW/index.md (hunk 11 above)
Comment (wmh) [inline lang/zh-TW/index.md]: 用「內部私有程式」呢？

File: lang/zh-TW/index.md (hunk 13 above)
Comment (wmh) [inline lang/zh-TW/index.md]: 共五處錯誤一併修改了。

File: lang/zh-TW/index.md (hunk 19 above)
Comment (wmh) [inline lang/zh-TW/index.md]: 我把第一句的「完全」去掉，變成「主版號為零的時候就是為了做快速開發」好像比較順一點。

File: lang/zh-TW/index.md (hunk 12 above)
Comment (wmh) [inline lang/zh-TW/index.md]: 另有四處「引入」也一併改為「加入」，有更通順 :+1:

**Comment by wmh** [comment]
Thank you very much @muan I learn a lot from this experience. I owe you one.

File: lang/zh-TW/index.md (hunk 5 above)
Comment (muan) [inline lang/zh-TW/index.md]: 「編譯版本資訊」很不錯！

File: lang/zh-TW/index.md (hunk 11 above)
Comment (muan) [inline lang/zh-TW/index.md]: 「主要的內部程式」怎麼樣..?

File: lang/zh-TW/index.md (hunk 19 above)
Comment (muan) [inline lang/zh-TW/index.md]: :+1:

**Comment by muan** [comment]
So throughout! :+1: Awesome!
There's a couple names pending, I made more suggestions. :smile:

File: lang/zh-TW/index.md (hunk 5 above)
Comment (dlackty) [inline lang/zh-TW/index.md]: 我也喜歡「版本編譯資訊」，聽起來通順、也不會和 version number 搞混。

Comment (wmh) [inline lang/zh-TW/index.md]: 「編譯版本資訊」vs「版本編譯資訊」 :smile:  :question:

Comment (muan) [inline lang/zh-TW/index.md]: 我覺得應該是「版本(build)編譯資訊(metadata)」？ :+1: @dlackty

File: lang/zh-TW/index.md (hunk 11 above)
Comment (wmh) [inline lang/zh-TW/index.md]: 這邊的 private code 指的應該是 API 內部的程式碼沒錯。感覺上「私有程式」和「內部程式」都可以，比較難和「主要的」想到關聯性 @dlackty 覺得呢？

**Comment by dlackty** [comment]
Looks pretty nice! :thumbsup: 
Really good job :)

**Comment by wmh** [comment]
@dlackty How would you suggest the translation of "private code" ? This is the last one we are not settled done yet.

File: lang/zh-TW/index.md (hunk 11 above)
Comment (dlackty) [inline lang/zh-TW/index.md]: 我覺得「內部程式」應該沒問題，我簡單查了一下、多數關於 Private Class / Methods 的翻譯都是翻成內部。
至於為什麼會提到「主要的」，我有漏掉什麼前後文嗎？

Comment (muan) [inline lang/zh-TW/index.md]: Lol 如果其他是翻成內部的話就內部吧
我是在想說例如GitHub的主程式和GitHub的API程式的分別..

Comment (dlackty) [inline lang/zh-TW/index.md]: :heart:  看起來已經有個共識了，那這個 pull request 大致上也差不多了？ @muan

File: lang/zh-TW/index.md [hunk 21]
```diff
@@ -0,0 +1,202 @@
...
+修訂號「必須 `MUST`」歸零。
+
+1. 先行版號「可以 `MAY`」被標注在修訂版之後，先加上一個連接號再加上一連串以句點
+分隔的標識符號來修飾。標識符號「必須 `MUST`」由 ASCII 碼的英數字和連接號
+[0-9A-Za-z-] 組成，且「禁止 `MUST NOT`」留白。數字型的標識符號「禁止 `MUST
+NOT`」在前方補零。先行版的優先級低於相關聯的標準版本。被標上先行版號則表示這個版
+本並非穩定而且可能無法達到相容的需求。範例：1.0​​.0-alpha、1.0.0-alpha.1、
+1.0.0-0.3.7、1.0.0-x.7.z.92。
+
+1. 編譯版號「可以 `MAY`」被標注在修訂版或先行版號之後，先加上一個連接號再加上一
```
Comment (wmh) [inline lang/zh-TW/index.md]: 自己發現錯誤，這邊應該是加號 :-1:

File: lang/zh-TW/index.md (hunk 11 above)
Comment (wmh) [inline lang/zh-TW/index.md]: ok, 我這邊都改完了。非常感謝兩位的協助 :bow:

**Comment by muan** [comment]
:+1::+1::+1:

@Haacked: It looks all good now! :blush: Do we need to put this in front of more people or will this be good to go?

**Comment by JuanitoFatas** [comment]
@wmh @muan Sorry I don't want to be mean, but there are still some spaces hiding in Chinese text.

**Comment by wmh** [comment]
@JuanitoFatas Would you please give some advice?

**Comment by JuanitoFatas** [comment]
@wmh Please see: https://github.com/wmh/semver.org/pull/1

**Comment by wmh** [comment]
@Haacked It's good to merge now.

**Comment by haacked** [comment]
Thanks!

**Comment by haacked** [comment]
thumbs-up-jackie-chan

**Pull request merged**